src/messaging/interface/session/response.ts

```diff
@@ -36,27 +36,40 @@ export async function receiveSessionRequest(
   { challenge, encryptionKeyUri: receiverEncryptionKeyUri }: ISessionRequest,
   encryptCallback: EncryptCallback,
   decryptCallback: DecryptCallback,
-  signCallback: SignCallback,
+  signers: SignerInterface[],
   {
-    resolveKey = Did.resolveKey,
+    dereferenceDidUrl = Did.dereference,
   }: {
-    resolveKey?: DidResolveKey
+    dereferenceDidUrl?: typeof Did.dereference
   } = {}
 ): Promise<{ session: ISession; sessionResponse: ISessionResponse }> {
   if (!didDocument.keyAgreement) {
     throw new Error('keyAgreement is necessary')
   }
-  const responseEncryptionKey = `${didDocument.uri}${didDocument.keyAgreement?.[0].id}` as DidResourceUri
+  const authenticationSigner = Signers.selectSigner<SignerInterface<Signers.DidPalletSupportedAlgorithms, DidUrl>>(
```

Comment:
Is this `KeyToolSigners` only for the test files? Maybe it should be used else where if that is the case.